src/common/managers/SceneManager.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+** EPITECH PROJECT, 2024
+** StellarForge
+** File description:
+** SceneManager.cpp
+*/
+
+#include "SceneManager.hpp"
+
+void SceneManager::addScene(const UUID &uuid, const std::shared_ptr<IScene> &scene, int position)
+{
+    if (_scenes.find(uuid) != _scenes.end()) {
+        throw ManagerException("Scene with UUID " + uuid.getUuidString() + " already exists.");
+    }
+    _scenes[uuid] = scene;
+
+    if (position < 0 || position >= _sceneOrder.size()) {
+        _sceneOrder.push_back(uuid);
+    } else {
+        _sceneOrder.insert(_sceneOrder.begin() + position, uuid);
```

Comment:
if position is >= _sceneOrder.size() I think insert does the same as push back but to check